pkg/controllers/job/job_controller_actions.go

```diff
@@ -464,12 +471,18 @@ func (cc *jobcontroller) syncJob(jobInfo *apis.JobInfo, updateStatus state.Updat
 		Conditions:          job.Status.Conditions,
 		RetryCount:          job.Status.RetryCount,
 	}
+	oldStatus := job.Status
+	job.Status = newStatus
 
 	if updateStatus != nil && updateStatus(&job.Status) {
 		job.Status.State.LastTransitionTime = metav1.Now()
```

Comment:
line 478-480 should be executed after `reflect.DeepEqual` check result is false, or line 483 check will always be true.

Reply:
``` go
if updateStatus != nil {
  updateStatus(&newStatus)
}

if reflect.DeepEqual(oldStatus, newStatus) {
  klog.xxx
  return nil
}else{
job.Status = newStatus
job.Status.State.LastTransitionTime = metav1.Now()
jobCondition = newCondition(job.Status.State.Phase, &job.Status.State.LastTransitionTime)
job.Status.Conditions = append(job.Status.Conditions, jobCondition)
}

```
and remove line 474-475

Reply:
done. Thks